### PR TITLE
fix: alive connections counter for graceful shutdown

### PR DESCRIPTION
**Pull request by DDtKey**
It's currently looks a bit fragile:
It's possible to spawn new future, but parallel task will finish before spawned task is polled. 
So this check will notify waiter & stop server:
```rs
if alive_connections.fetch_sub(1, Ordering::SeqCst) == 1 {
    notify.notify_one();
}
```

It's hard to test, however sometimes we're encountering `connection reset` errors on client end, while server side almost immediately logged correct graceful shutdown & server stopped (we use large enough timeout, so it's not the case)

I believe it's more reliable & safe to increase counter before spawning the task of serving.

### CODE REVIEW

File: poem/src/server.rs [hunk 1]
```diff
@@ -135,9 +135,8 @@ where
                         let notify = notify.clone();
                         let timeout_notify = timeout_notify.clone();
 
+                        alive_connections.fetch_add(1, Ordering::SeqCst);
```
Comment (DDtKey) [inline poem/src/server.rs:138]: Btw, I see `SeqCst` is used here and it's strict. 
Probably it should be refactored to `Acquire/Release` 🤔

Comment (sunli829) [reply poem/src/server.rs:138]: Could you help me with this? 🙂

Comment (DDtKey) [reply poem/src/server.rs:138]: Sure!
As separate PR or this one?

Comment (sunli829) [reply poem/src/server.rs]: thanks 🙂

**Pull request merged**